kotlin-android/app/src/main/java/com/mongodb/tasktracker/LoginActivity.kt

```diff
@@ -47,14 +51,17 @@ class LoginActivity : AppCompatActivity() {
 
 
         if (createUser) {
+            // register a user using the Realm App we created in the TaskTracker class
             taskApp.emailPasswordAuth.registerUserAsync(username, password) {
+                // while this operation completes, disable the buttons to login or create a new account
                 binding.buttonCreate.isEnabled = true
                 binding.buttonLogin.isEnabled = true
                 if (!it.isSuccess) {
                     onLoginFailed("Could not register user.")
                     Log.v(TAG(), "Error: ${it.error}")
                 } else {
                     Log.v(TAG(), "Successfully registered user.")
+                    // when the account has been created successfully, log in to the account
                     login(false)
                 }
             }
```

Comment:
On that note, why not disable inputs in the else block as well? (Which means you could just disable inputs right above the opening `if`)